### ISSUE TITLE
Can't add basic authentication to Nginx

### ISSUE THREAD

**Issue by tms320** [sameersbn/docker-redmine#365]
I want to add basic authentication to restrict access to Redmine.
My steps:
1. I run container using the following command (I use external MySQL server):
docker run --name=redmine -it --rm -d \\
  -p 3000:80 \\
  --env='DB_ADAPTER=mysql2' \\
  --env='DB_HOST=192.168.2.1' --env='DB_NAME=redmine_production' \\
  --env='DB_USER=redmine' --env='DB_PASS=password' \\
  --volume=/srv/docker/redmine/data:/home/redmine/data \\
  --volume=/srv/docker/redmine/logs:/var/log/redmine \\
  sameersbn/redmine
The container started successfully.
2. I "attach" to the running container executing shell:
`docker exec -i -t redmine bash`
So, now I am "inside" container.
3. I install 'Midnight Commander':
apt update
apt install mc
4. I modify config file '/etc/nginx/sites-enabled/redmine' adding the following lines:
auth_basic "Enter Login/Password";
auth_basic_user_file /home/redmine/data/redmine_passwd;
to the 'server' section.
5. I create password file '/home/redmine/data/redmine_passwd' (using online generator).
6. I reload Nginx using command:
`service nginx reload`
7. Basic authentication now works fine. When I try to open Redmine web page I get prompt to enter login/password.
8. Now I have to save changes in my container as new docker image.
I exit from container session (see step 2 above). Container continues working.
I commit changes using command:
`docker commit 09bd66ab6aa2 redmine2`
So I get new image 'redmine2'.
9. I stop original container:
`docker stop redmine`
10. I run new container based on new image 'redmine2':
docker run --name=redmine -it --rm -d \\
  -p 3000:80 \\
  --env='DB_ADAPTER=mysql2' \\
  --env='DB_HOST=192.168.2.1' --env='DB_NAME=redmine_production' \\
  --env='DB_USER=redmine' --env='DB_PASS=password' \\
  --volume=/srv/docker/redmine/data:/home/redmine/data \\
  --volume=/srv/docker/redmine/logs:/var/log/redmine \\
  redmine2
11. The container started successfully. But when I try to open Redmine web page it is opened without prompt to enter login/password.
I "attach" to the running container (like step 2 above) and check the files I think I modified.
I run 'Midnight Commander' and saw that the config file '/etc/nginx/sites-enabled/redmine' remains unchanged (restored to original version). WHY?
The password file '/home/redmine/data/redmine_passwd' remains as it was created.
12. I tried to map directory '/etc/nginx/sites-enabled' to host using option:
`--volume /srv/docker/redmine/nginx/sites-enabled:/etc/nginx/sites-enabled`
but it didn't help. The config file '/etc/nginx/sites-enabled/redmine' remains original.
How can I solve my problem?
Thanks.

**Comment by jcormier** [comment]
The nginx config file is copied during startup from /etc/docker-redmine/runtime/config/nginx depending on if you select ssl or not.

I mapped my custom nginx script like so ```--volume ./config/nginx/redmine_3.3.1:/etc/docker-redmine/runtime/config/nginx/redmine:ro```

**Comment by tms320** [comment]
Thanks for information.
I tried to map container's directory '/etc/docker-redmine/runtime/config/nginx/' to host using the option:
--volume=/srv/docker/redmine/nginx:/etc/docker-redmine/runtime/config/nginx
But the container stops almost immediately after I run it. I couldn't find any error messages in container's logs why it can't to launch.
I tried to add ':ro' at the end of option line - it didn't help (by the way, what ':ro' means?)
